django/contrib/auth/management/commands/createsuperuser.py

```diff
@@ -121,22 +125,36 @@ def handle(self, *args, **options):
                                 field.remote_field.field_name,
                             ) if field.remote_field else '',
                         ))
-                        user_data[field_name] = self.get_input_data(field, message)
+                        input_value = self.get_input_data(field, message)
+                        user_data[field_name] = input_value
+                        fake_user_data[field_name] = input_value
+
+                        # Wrap any foreign keys in fake model instances
+                        if field.remote_field:
+                            fake_user_data[field_name] = field.remote_field.model(input_value)
 
                 # Get a password
                 while password is None:
-                    if not password:
-                        password = getpass.getpass()
-                        password2 = getpass.getpass(force_str('Password (again): '))
-                        if password != password2:
-                            self.stderr.write("Error: Your passwords didn't match.")
-                            password = None
-                            continue
+                    password = getpass.getpass()
+                    password2 = getpass.getpass(force_str('Password (again): '))
+                    if password != password2:
+                        self.stderr.write("Error: Your passwords didn't match.")
+                        password = None
+                        continue
+
                     if password.strip() == '':
                         self.stderr.write("Error: Blank passwords aren't allowed.")
                         password = None
                         continue
 
+                    try:
+                        fake_user = self.UserModel(**fake_user_data)
+                        validate_password(password2, fake_user)
+                    except exceptions.ValidationError as err:
+                        self.stderr.write(', '.join(err.messages))
+                        password = None
+                        continue
```

Comment:
I can't establish from the diff, what's happening differently from the same logic addition in `changepassword.py` that means this has a `continue` where the other one does not?